packages/fxa-auth-server/scripts/stripe-products-and-plans-to-firestore-documents/stripe-product-and-plans-converter.ts

```diff
@@ -0,0 +1,382 @@
+  }
+
+  // TODO: Better method name and JSDoc
+  /**
+   * For each Stripe Product (only the product with `productId` else all products)
+   *  - Initialize an empty object to represent a productConfig Firestore doc
+   *  - Copy any product metadata to the productConfig
+   *  - Check if a document already exists for this Stripe product ID and get its doc ID
+   *  - For each Stripe Plan
+   *    - Copy any plan locale metadata to productConfig.locales
+   *    - Initialize an empty object to represent a planConfig Firestore doc
+   *    - Copy any other plan metadata to the planConfig
+   *    - Check if a document already exists for this Stripe plan ID and get its doc ID
+   *    - Update or add the planConfig into the PlanConfig collection in Firestore
+   *  - Update or add the productConfig into the ProductConfig collection in Firestore
+   */
+  async convert({
+    productId,
+    isDryRun,
+  }: {
+    productId: string;
+    isDryRun: boolean;
+  }): Promise<void> {
+    this.log.info('StripeProductsAndPlansConverter.convertBegin', {
+      productId,
+      isDryRun,
+    });
+    await this.paymentConfigManager.load();
+    this.paymentConfigManager.startListeners();
+    const params = productId ? { ids: [productId] } : {};
+    for await (const product of this.stripeHelper.stripe.products.list(
+      params
+    )) {
+      const productConfig = this.stripeProductToProductConfig(product);
+      // We need a productConfigId to store a planConfig; this may require
+      // storing a new productConfig with `locales: {}` and updating it after
+      // we've iterated through the Stripe Plans.
+      const existingProductConfigId =
+        await this.paymentConfigManager.getDocumentIdByStripeId(product.id);
+      let productConfigId;
+      if (existingProductConfigId) {
+        productConfigId = existingProductConfigId;
+      } else {
+        productConfigId = await this.paymentConfigManager.storeProductConfig(
+          productConfig
+        );
+      }
+      for await (const plan of this.stripeHelper.stripe.plans.list({
+        product: product.id,
+      })) {
+        this.stripePlanLocalesToProductConfigLocales(plan);
+        productConfig.locales = {
+          ...productConfig.locales,
+          ...this.stripePlanLocalesToProductConfigLocales(plan),
+        };
+        const planConfig = this.stripePlanToPlanConfig(plan, productConfigId);
+        // If a planConfig doc already exists, update it rather than creating
+        // a new doc
+        const existingPlanConfigId =
+          await this.paymentConfigManager.getDocumentIdByStripeId(plan.id);
+        const planConfigId = existingPlanConfigId
+          ? await this.paymentConfigManager.storePlanConfig(
+              planConfig,
+              productConfigId,
+              existingPlanConfigId
+            )
+          : await this.paymentConfigManager.storePlanConfig(
+              planConfig,
+              productConfigId
+            );
```

Comment:
I conditionally pass a second argument into `PaymentConfigManager` if `existingPlanConfigId` exists. Is there a more concise way of doing this?

Reply:
Chatted with Barry; I'll just update the type on the argument into `storePlanConfig` and `storeProductConfig` to be `string | null`.